Add codespell: config, workflow (to alert when new typos added) and get typos fixed

yarikoptic opened the pull request:
This is my way to contribute back to projects I touch to use or told about (thus popular).

codespell: https://github.com/codespell-project/codespell to discover more


I spotted some code changes - but all looked good to be changed, but please review more.

I ignored 'unsecure'  in a config setting `connection.unsecure` to not cause disturbance. 

You can annotate lines to be ignored with e.g. `/* pragma: codespell-ignore */` so you could even add suggestions right here in the diff as suggestions to expedite reversion back if needed.

yarikoptic committed:
Add github action to codespell master on push and PRs

yarikoptic committed:
Add rudimentary codespell config with custom ignores

yarikoptic committed:
ignore connection.unsecure config

yarikoptic committed:
ignore perm=fle

yarikoptic committed:
ignore any line with .*codespell-ignore.*

yarikoptic committed:
ignore .resx files

yarikoptic committed:
codespell ignore a line in the tests

yarikoptic committed:
ignore .plist files and a case in a test

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/iterate-ch/cyberduck?pullRequest=15437) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/iterate-ch/cyberduck?pullRequest=15437) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/iterate-ch/cyberduck?pullRequest=15437) it.</sub>

yarikoptic commented:
macos-13 is failing also in https://github.com/iterate-ch/cyberduck/actions/runs/7120060984 for another PR

dkocher commented:
Our [documentation](https://github.com/iterate-ch/docs) would benefit from this build action a lot more.

dkocher commented:
Generated code from swagger-codegen should be excluded from spelling checks.

yarikoptic commented:
is it the `swagger/` directories?

what is it generated from since it seems to have unique to it typos

```shell
❯ git grep beeing
storegate/src/main/java/ch/cyberduck/core/storegate/io/swagger/client/api/WebDavPasswordApi.java:   * Remove a WebDAV password to stop it from beeing used.
storegate/src/main/java/ch/cyberduck/core/storegate/io/swagger/client/api/WebDavPasswordApi.java:   * Remove a WebDAV password to stop it from beeing used.
```

dkocher commented:
> is it the `swagger/` directories?
> 
> what is it generated from since it seems to have unique to it typos
> 
> ```shell
> ❯ git grep beeing
> storegate/src/main/java/ch/cyberduck/core/storegate/io/swagger/client/api/WebDavPasswordApi.java:   * Remove a WebDAV password to stop it from beeing used.
> storegate/src/main/java/ch/cyberduck/core/storegate/io/swagger/client/api/WebDavPasswordApi.java:   * Remove a WebDAV password to stop it from beeing used.
> ```

The data types are generated from an API specification (that includes the typos) outside the scope of this project.

yarikoptic committed:
ignore swagger folders

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout fixing typo automagically but ignoring the failure due to ambigous typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w || :",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] Do interactive fixing of  leftover ambigous typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 2",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
ok, ignored `swagger`, rebased and reran typos fixing.
